iree/compiler/Codegen/LLVMGPU/LLVMGPUTileAndDistribute.cpp

```diff
@@ -408,7 +363,15 @@ struct LLVMGPUTileAndDistributePass
             funcOp, std::move(promotionCanonicalization));
       }
 
+      LLVM_DEBUG({
+        llvm::dbgs() << "After promotion:";
+        funcOp.dump();
+      });
+
       {
+        auto workgroupSize = llvm::to_vector<4>(llvm::map_range(
```

Comment:
So does this basically replace "Applying WgTileSize to Fop using concretizewgsizeop + materializestaticlaunchinfo" ?

Reply:
I see, and I guess most of the logic gets transferred over to "populateTilingReductionPatterns"?

Reply:
I am not sure I follow the comment exactly. But the intent here is that this pass is doing just tile and distribute given the configuration. The `LLVMGPULowerExecutableTargetPass` first sets the configuration and then calls the cuda lowering pipeline.
`populateTilingReductionPatterns` is just reading the tile sizes from the attribute on the op.

Reply:
hmm, I guess the main question/confusion I had is in tileAndDistribute we used to have ConcretizeWgSizeOp to get tileSize for the WG and materializeStaticLaunch information to compute WG sizes, so when we remove these two from tileAndDistribute, what is the new flow/process to achieve what these two were doing?

Reply:
The `materializeStaticLaunch` method itself calls `SetWorkgroupSizePattern` which is redundant with `ConcretizeWgSizeOp`. The decision to find the configuration is moved into `LLVMGPULowerExecutableTargetPass`. Within that, the call to `initGPULaunchConfiguration` sets the information about tile size and workgroup size. The `SetWorkgroupSizePattern` uses that information to set the "workload per workgroup" (which is same as tile size). So now by the time tile and distribute is called, this is all already done.

Reply:
I see I see, that makes sense! thanks for the explanation Mahesh :)